### DISCUSSION TITLE
Constraints are inconsistent

### DISCUSSION THREAD

**Discussion by suitangxiaoyi** [yalmip/YALMIP#1445] [General discussion] [answered]
Hi everybody,
When I solve a problem, It gives some error as follows:
`+ Solver chosen : CPLEX-IBM
+ Processing objective function
+ Processing constraints
+ Calling CPLEX
Matlab Toolbox parameter "Display" found.
Version identifier: 12.10.0.0 | 2019-11-26 | 843d4de2ae
CPXPARAM_Simplex_Display                         2
CPXPARAM_Barrier_Display                         2
Constraints 'c30338' and 'c30345' are inconsistent.
Presolve time = 0.01 sec. (14.42 ticks)

Root node processing (before b&c):
  Real time             =    0.01 sec. (14.67 ticks)
Parallel b&c, 32 threads:
  Real time             =    0.00 sec. (0.00 ticks)
  Sync time (average)   =    0.00 sec.
  Wait time (average)   =    0.00 sec.
                          ------------
Total (root+branch&cut) =    0.01 sec. (14.67 ticks)`

I only have 8670 Constraints, so I couldn't find Constraints 'c30338' and 'c30345'. What is causing it and how can it be fixed?

**Comment by johanlofberg** [comment, marked as answer]
https://yalmip.github.io/debugginginfeasible

**Reply by suitangxiaoyi** [reply]
# Get rid of the objective function. 
`options=sdpsettings('verbose',2,'solver','cplex','showprogress',1);` becomes `optimize(Constraints)`
It throws the same error report：
`Matlab Toolbox parameter "Display" found.
警告: Using Matlab Toolbox parameters in the CPLEX Matlab API is deprecated. They will be treated as unknown parameters in the future.
Version identifier: 12.10.0.0 | 2019-11-26 | 843d4de2ae
CPXPARAM_MIP_Display                             1
Constraints 'c30338' and 'c30345' are inconsistent.
Presolve time = 0.02 sec. (14.58 ticks)
ans = 包含以下字段的 struct:
    yalmipversion: '20230622'
    matlabversion: '24.1.0.2537033 (R2024a)'
       yalmiptime: 54.4718
       solvertime: 0.6172
             info: 'Infeasible problem ([learn to debug](https://127.0.0.1:31515/toolbox/matlab/editor/application/yalmip.github.io/debugginginfeasible)) (CPLEX-IBM)'
          problem: 1`
# Try create sub-components
I tried to do this, but unfortunately I found that I couldn't link Constraints 'c30338' and 'c30345' to any of the original constraints, so what should I do in this case?
# Re-order group constraints
I try to run every constraints group with rid of the objective function. Eventually, discover the specific location where the constraint set is located. But I'm still not sure how to go about solving this problem.

```
Constraints=[Constraints,sum(sum([x_pij_a(node_in,opt_num);x_pij_b(node_in,opt_num);x_pij_c(node_in,opt_num)]-...
                real(r_ij(3*node_in-2:3*node_in,:))*[x_Iij_square_a(node_in,opt_num);x_Iij_square_b(node_in,opt_num);x_Iij_square_c(node_in,opt_num)]))+...
               +[p_Solar_a(k,opt_num);p_Solar_b(k,opt_num);p_Solar_c(k,opt_num)]+...
                [p_Wind_a(k,opt_num);p_Wind_b(k,opt_num);p_Wind_c(k,opt_num)]-[p_Load_a(k,opt_num);p_Load_b(k,opt_num);p_Load_c(k,opt_num)]...
                ==sum(sum([x_pij_a(node_out,opt_num);x_pij_b(node_out,opt_num);x_pij_c(node_out,opt_num)]))];
            Constraints=[Constraints,sum(sum([x_qij_a(node_in,opt_num);x_qij_b(node_in,opt_num);x_qij_c(node_in,opt_num)]-...
                imag(r_ij(3*node_in-2:3*node_in,:))*[x_Iij_square_a(node_in,opt_num);x_Iij_square_b(node_in,opt_num);x_Iij_square_c(node_in,opt_num)]))
                -[p_Load_a(k,opt_num);p_Load_b(k,opt_num);p_Load_c(k,opt_num)]...
                ==sum(sum([x_qij_a(node_out,opt_num);x_qij_b(node_out,opt_num);x_qij_c(node_out,opt_num)]))];

```
Next, I'm going to try to make this infeasible constraint feasible.

**Reply by johanlofberg** [reply]
You simply have to start commenting out individual constraints, creating smaller model, etc etc, and if infeasibility remains you know it can be removed from consideration and keep removing constraints. If feasibility is recovered, you keep it as it might be the culprit. Simplifying the model like this shuld in the end lead to a model with only a few constraints remaining from where you will have to understand why they are trivially inconsistent
